packages/cli-kit/src/common/result.test.ts

```diff
@@ -0,0 +1,59 @@
+import {err, ok} from './result.js'
+import {describe, expect, it} from 'vitest'
+import {fail} from 'node:assert'
+
+describe('ok', () => {
+  it('create ok with value', () => {
+    // When
+    const result = ok(123)
+
+    // Then
+    if (result.isErr()) {
+      fail('Should return an Ok result')
+    } else {
+      expect(result.value).toEqual(123)
+    }
```

Comment:
Why not use expects for the irErr too?
```suggestion
      expect(result.isErr()).toBeFalsy()
      expect(result.value).toEqual(123)
```

Reply:
Using the method `isErr` infers the type of the `result` (Ok or Err) so either `value` or `error` could be accessed. `expect` could be used but evaluation of the method `isErr` is mandatory to access one of those values directly